### PR TITLE
fixing warnings in tests code, part6

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: test/test_buffer.cpp [hunk 1]
```diff
@@ -152,7 +152,7 @@ void free_buffer(char* m)
 
 char* allocate_buffer(int size)
 {
-	char* mem = (char*)std::malloc(size);
+	char* mem = static_cast<char*>(std::malloc(std::size_t(size)));
```
Comment (arvidn) [inline test/test_buffer.cpp]: I wonder if there's a reason not to use ``new char[size]`` here instead, and avoid the casts

File: test/test_dht.cpp [hunk 2]
```diff
@@ -2583,11 +2594,11 @@ TORRENT_TEST(traversal_done)
...
 	// invert the ith most significant byte so that the test nodes are
 	// progressively closer to the target item
-	for (int i = 0; i < num_test_nodes; ++i)
+	for (std::size_t i = 0; i < num_test_nodes; ++i)
 		nodes[i].id[i] = ~nodes[i].id[i];
 
 	// add the first k nodes to the subject's routing table
-	for (int i = 0; i < 8; ++i)
+	for (std::size_t i = 0; i < 8; ++i)
 		t.dht_node.m_table.add_node(nodes[i]);
```
Comment (arvidn) [inline test/test_dht.cpp]: could this be a range-for over ``nodes``?

File: test/test_dht.cpp [hunk 3]
```diff
@@ -3305,7 +3316,7 @@ TORRENT_TEST(invalid_error_msg)
 	node.incoming(node.m_sock, m);
 
 	bool found = false;
-	for (int i = 0; i < int(observer.m_log.size()); ++i)
+	for (std::size_t i = 0; i < observer.m_log.size(); ++i)
```
Comment (arvidn) [inline test/test_dht.cpp]: this looks like it could be a range-for loop over ``observer.m_log``.

**Comment by aldenml** [comment]
sure, I will use the for-loop and `new char[size]`

File: test/test_buffer.cpp (hunk 1 above)
Comment (arvidn) [reply test/test_buffer.cpp]: if the ``chained_buffer`` frees with ``free``, this would still have to be ``malloc``

Comment (aldenml) [reply test/test_buffer.cpp]: OK

Comment (aldenml) [reply test/test_buffer.cpp]: yes, `std::free` is used with the `free_buffer` above, no change here

File: test/test_dht.cpp (hunk 2 above)
Comment (aldenml) [reply test/test_dht.cpp]: can't modify this one because nodes is of `9` elements, but using it in others

File: test/test_dht.cpp (hunk 3 above)
Comment (aldenml) [reply test/test_dht.cpp]: done

**Pull request merged**